smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -179,36 +300,47 @@ private void generateOperationSerializer(GenerationContext context, OperationSha
         return queryBindings;
     }
 
-    private String getInputValue(
+    protected String getInputValue(
+            GenerationContext context,
             Location bindingType,
-            OperationShape operation,
+            Shape shape,
             MemberShape member,
             Shape target
     ) {
         String memberName = member.getMemberName();
+        SymbolProvider symbolProvider = context.getSymbolProvider();
 
-        if (target instanceof StringShape) {
+        if (target instanceof StringShape || target instanceof DocumentShape) {
             return "input." + memberName;
         } else if (target instanceof BooleanShape || target instanceof NumberShape) {
             // Just toString on the value.
             return "input." + memberName + ".toString()";
         } else if (target instanceof TimestampShape) {
             return getTimestampInputParam(member, bindingType);
         } else if (target instanceof BlobShape) {
-            // base64 encode
-            // TODO: fixme (how do we base64 encode?)
-            throw new UnsupportedOperationException("Not yet implemented");
+            return getBlobInputParam(member, bindingType);
         } else if (target instanceof CollectionShape) {
-            // TODO: fixme
-            throw new UnsupportedOperationException("Not yet implemented");
+            return getCollectionInputParam(context, bindingType, shape, member, target);
+        } else if (target instanceof StructureShape || target instanceof UnionShape || target instanceof MapShape) {
+            Symbol symbol = symbolProvider.toSymbol(target);
+            String value = ProtocolGenerator.getSerdeFunctionName(symbol, getName(), true) + "(";
+            // Collections map over the input entries, so use that.
+            // TODO See collection cleanup note
+            if (shape instanceof CollectionShape) {
```

Comment:
This should be removed I think. Line 322 would cause this to never be called.

Reply:
This is checking the container being generated for, not the target type like L322. This handles swapping the input parameter on things like this:

```ts
  input.map(entry =>
    serializeAws_restJson1_1Foo(entry, context)
  );
```